streaming/src/main/scala/org/apache/spark/streaming/ui/AllBatchesTable.scala

```diff
@@ -17,39 +17,122 @@
 
 package org.apache.spark.streaming.ui
 
-import scala.xml.Node
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets.UTF_8
+import javax.servlet.http.HttpServletRequest
+
+import scala.xml.{Node, Unparsed}
+
+import org.apache.spark.ui.{PagedDataSource, PagedTable, UIUtils => SparkUIUtils}
+
+private[ui] class StreamingBatchPagedTable(
+    request: HttpServletRequest,
+    parent: StreamingTab,
+    batchInterval: Long,
+    batchData: Seq[BatchUIData],
+    streamingBatchTag: String,
+    basePath: String,
+    subPath: String,
+    parameterOtherTable: Iterable[String],
+    pageSize: Int,
+    sortColumn: String,
+    desc: Boolean) extends PagedTable[BatchUIData] {
+
+  override val dataSource = new StreamingBatchTableDataSource(batchData, pageSize, sortColumn, desc)
+  private val parameterPath = s"$basePath/$subPath/?${parameterOtherTable.mkString("&")}"
+  private val firstFailureReason = getFirstFailureReason(batchData)
+
+  override def tableId: String = streamingBatchTag
+
+  override def tableCssClass: String =
+    "table table-bordered table-condensed table-striped " +
+      "table-head-clickable table-cell-width-limited"
+
+  override def pageLink(page: Int): String = {
+    val encodedSortColumn = URLEncoder.encode(sortColumn, UTF_8.name())
+    parameterPath +
+      s"&$pageNumberFormField=$page" +
+      s"&$streamingBatchTag.sort=$encodedSortColumn" +
+      s"&$streamingBatchTag.desc=$desc" +
+      s"&$pageSizeFormField=$pageSize"
```

Comment:
Don't we need to add `#tableHeaderId` here?
https://github.com/apache/spark/blob/67b644c3d74b0587dd5d498b903383ac4de932fe/sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala#L259